vms/platformvm/block/builder/builder.go

```diff
@@ -308,6 +292,85 @@ func buildBlock(
 	)
 }
 
+func packBlockTxs(mempool mempool.Mempool, backend *txexecutor.Backend, manager blockexecutor.Manager, timestamp time.Time) ([]*txs.Tx, error) {
+	preferredID := manager.Preferred()
+	stateDiff, err := state.NewDiff(preferredID, manager)
+	if err != nil {
+		return nil, err
+	}
+
+	changes, err := txexecutor.AdvanceTimeTo(backend, stateDiff, timestamp)
+	if err != nil {
+		return nil, err
+	}
+	changes.Apply(stateDiff)
+	stateDiff.SetTimestamp(timestamp)
+
+	var (
+		blockTxs      []*txs.Tx
+		inputs        set.Set[ids.ID]
+		remainingSize = targetBlockSize
+	)
+
+	for {
+		tx, exists := mempool.Peek()
+		if !exists {
+			break
+		}
+		txSize := len(tx.Bytes())
+		if txSize > remainingSize {
+			break
+		}
+		mempool.Remove([]*txs.Tx{tx})
+
+		// Invariant: [tx] has already been syntactically verified.
+
+		txDiff, err := state.NewDiffOn(stateDiff)
+		if err != nil {
+			return nil, err
+		}
+
+		executor := &txexecutor.StandardTxExecutor{
+			Backend: backend,
+			State:   txDiff,
+			Tx:      tx,
+		}
+
+		err = tx.Unsigned.Visit(executor)
+		if err != nil {
+			txID := tx.ID()
+			mempool.MarkDropped(txID, err)
+			continue
+		}
+
+		if inputs.Overlaps(executor.Inputs) {
+			txID := tx.ID()
+			mempool.MarkDropped(txID, blockexecutor.ErrConflictingBlockTxs)
+			continue
+		}
+		err = manager.VerifyUniqueInputs(preferredID, executor.Inputs)
+		if err != nil {
+			txID := tx.ID()
+			mempool.MarkDropped(txID, err)
+			continue
+		}
+		inputs.Union(executor.Inputs)
+
+		txDiff.AddTx(tx, status.Committed)
+		err = txDiff.Apply(stateDiff)
+		if err != nil {
+			txID := tx.ID()
+			mempool.MarkDropped(txID, err)
+			continue
+		}
```

Comment:
I think we should return this error. If `txDiff.Apply` errors - we are not guaranteed that `stateDiff` is unmodified... Also it is just unexpected for this to ever error.

Reply:
Returned